File: ClientApp/src/app/Module/Shared/enum/AllStatus.ts
export enum orderDetailStatus {
    APPROVED = 'APPROVED',
    REJECTED = 'REJECTED',
    RETURNED = 'RETURNED',
    PARTIALRETURN = 'PARTIALRETURN',
}

export enum orderStatus {
    PENDING = 'PENDING',
    APPROVED = 'APPROVED',
    REJECTED = 'REJECTED',
    DRAFT  = 'DRAFT',
    SUBMIT = 'SUBMIT',
    WAITAPPROVE = 'WAITAPPROVE',
    CANCELLED = 'CANCELLED',
}

export const orderDetailStatusLabel: Record<orderDetailStatus, string> = {
    [orderDetailStatus.APPROVED]: 'Approved',
    [orderDetailStatus.REJECTED]: 'Rejected',
    [orderDetailStatus.RETURNED]: 'Returned',
    [orderDetailStatus.PARTIALRETURN]: 'Partial Return',
};

export const orderStatusLabel: Record<orderStatus, string> = {
    [orderStatus.PENDING]: 'Pending',
    [orderStatus.APPROVED]: 'Approved',
    [orderStatus.REJECTED]: 'Rejected',
    [orderStatus.DRAFT]: 'Draft',
    [orderStatus.SUBMIT]: 'Submit',
    [orderStatus.WAITAPPROVE]: 'Wait Approve',
    [orderStatus.CANCELLED]: 'Cancelled',
};

File: HomeWork.Domain/Enums/AllStatus.cs
namespace HomeWork.Domain.Enums;

public class EnumOrderStatus
{
    public const string PENDING = "PENDING";
    public const string APPROVED = "APPROVED";
    public const string REJECTED = "REJECTED";
    public const string DRAFT = "DRAFT";
    public const string SUBMIT = "SUBMIT";
    public const string WAITAPPROVE = "WAITAPPROVE";
    public const string CANCELLED = "CANCELLED";
}

public class EnumOrderDetailStatus
{
    public const string APPROVED = "APPROVED";
    public const string REJECTED = "REJECTED";
    public const string RETURNED = "RETURNED";
    public const string PARTIALRETURN = "PARTIALRETURN";
}
